pvlib/mlfm.py

```diff
@@ -0,0 +1,803 @@
+
+        return dstack
+
+    # subtractive losses without series and shunt resistance effects
+    # find factor to transform multiplicative to subtractive losses
+    # correction factor to scale losses to keep 1/ff --> pr_dc
+
+    prod = inv_ff * (
+        dnorm['i_sc'] * dnorm['i_mp'] *
+        dnorm['v_mp'] * dnorm['v_oc']
+    )
+
+    tot = inv_ff + (
+        dnorm['i_sc'] + dnorm['i_mp'] +
+        dnorm['v_mp'] + dnorm['v_oc'] - 4
+    )
+
+    corr = (inv_ff - prod) / (inv_ff - tot)
+
+    # put mlfm values in a stack from pr_dc (bottom) to 1/ff_ref (top)
+    dstack['pr_dc'] = + dnorm['pr_dc']  # initialise
+    dstack['i_sc'] = -(dnorm['i_sc'] - 1) * corr
+    dstack['i_mp'] = -(dnorm['i_mp'] - 1) * corr - gap/2
+    dstack['i_v'] = gap
+    dstack['v_mp'] = -(dnorm['v_mp'] - 1) * corr - gap/2
+    dstack['v_oc'] = -(dnorm['v_oc'] - 1) * corr
+
+    dstack['temp_module_corr'] = (
+        - (dnorm['v_oc'] - dnorm['v_oc_temp_corr']) * corr)
+
+    return dstack
+
+
+def mlfm_6(dmeas, c_1, c_2, c_3, c_4, c_5=0., c_6=0.):
+    r'''
+    Predict normalised LFM values from data in ``dmeas``.
+
+    The normalized LFM values are given by
+
+    .. math::
+
+        c_1 + c_2 (T_m - 25) + c_3 \log10(G_{POA}) + c_4 G_{POA}
+        + c_5 WS + c_6 / G_{POA}
+
+    where :math:`G_{POA}` is global plane-of-array (POA) irradiance in kW/m2,
+    :math:`T_m` is module temperature in C and :math:`WS` is wind speed in
+    m/s.
+
+    Parameters
+    ----------
+    dmeas : DataFrame
+        Must include columns:
+
+        * `'poa_global_kwm2'` global plane of array irradiance [kW/m^2]
+        * `'temp_module'` module temperature [C]
+
+        May include optional column:
+
+        * `'wind_speed'` wind speed [m/s].
+
+    c_1 : float
+        Constant term in model.
+    c_2 : float
+        Temperature coefficient in model. [1/C]
+    c_3 : float
+        Coefficient for low light log irradiance drop.
+    c_4 : float
+        Coefficient for high light linear irradiance drop.
+    c_5 : float, default 0
+        Coefficient for wind speed dependence.
+    c_6 : float, default 0
+        Coefficient for dependence on inverse irradiance.
+
+    Returns
+    -------
+    mlfm_6 : Series
+        Predicted values.
+
+    References
+    ----------
+    .. [1] Steve Ransome (SRCL) and Juergen Sutterlueti (Gantner Instruments)
+       "Quantifying Long Term PV Performance and Degradation under Real Outdoor
+       and IEC 61853 Test Conditions Using High Quality Module IV Measurements"
+       36th EU PVSEC, Marseille, France. September 2019
+     '''
+    mlfm_out = c_1 + c_2 * (dmeas['temp_module'] - T_STC) + \
+        c_3 * np.log10(dmeas['poa_global_kwm2']) + \
+        c_4 * dmeas['poa_global_kwm2'] + c_6 / dmeas['poa_global_kwm2']
+    if 'wind_speed' in dmeas.columns:
+        mlfm_out += c_5 * dmeas['wind_speed']
+    return mlfm_out
+
+
+def mlfm_fit(data, var_to_fit):
+    '''
+    Fit MLFM to data.
```

Comment:
Newbies like me might benefit from a mention of what variables are appropriate to fit with this function.  `pr_dc`, others?